grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckServiceTest.java

```diff
@@ -0,0 +1,363 @@
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.requestTimeoutMillis(3000);
+            sb.decorator(LoggingService.newDecorator());
+            sb.service(GrpcService.builder()
+                                  .addService(new TestServiceImpl())
+                                  .grpcHealthCheckService(service)
+                                  .build());
+        }
+    };
+
+    @BeforeEach
+    void setUp() {
+        serveHealth.setHealthy(true);
+    }
+
+    @Test
+    void check() {
+        final HealthBlockingStub client = Clients.newClient(
+                server.httpUri(GrpcSerializationFormats.PROTO),
+                HealthBlockingStub.class);
+        HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        HealthCheckResponse response = client.check(request);
+        assertThat(response.getStatus()).isEqualTo(ServingStatus.SERVING);
+
+        request = HealthCheckRequest.newBuilder()
+                                    .setService("com.linecorp.armeria.grpc.testing.TestService")
+                                    .build();
+        response = client.check(request);
+        assertThat(response.getStatus()).isEqualTo(ServingStatus.SERVING);
+
+        final StatusRuntimeException exception = assertThrows(StatusRuntimeException.class, () -> client.check(
+                HealthCheckRequest.newBuilder()
+                                  .setService("com.linecorp.armeria.grpc.testing.NotFoundTestService")
+                                  .build()));
+        assertThat(exception.getStatus().getCode()).isEqualTo(Status.NOT_FOUND.getCode());
+        assertThat(exception.getMessage()).isEqualTo(
+                "NOT_FOUND: The service name(com.linecorp.armeria.grpc.testing.NotFoundTestService) " +
+                "is not registered in this service");
+    }
+
+    @Test
+    void watch() throws Exception {
+        final HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        final HealthStub client = Clients.newClient(
+                server.httpUri(GrpcSerializationFormats.PROTO),
+                HealthStub.class);
+        final StreamRecorder<HealthCheckResponse> recorder = StreamRecorder.create();
+        client.watch(request, recorder);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(false);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(true);
+        TimeUnit.SECONDS.sleep(1);
+        recorder.onCompleted();
+        final List<ServingStatus> responses = recorder.getValues()
+                                                      .stream()
+                                                      .map(HealthCheckResponse::getStatus)
+                                                      .collect(Collectors.toList());
+        assertThat(responses).containsExactlyElementsOf(
+                ImmutableList.of(ServingStatus.SERVING, ServingStatus.NOT_SERVING, ServingStatus.SERVING));
```

Comment:
nit; no need for the extra allocation

```suggestion
        assertThat(responses)
                .containsExactly(ServingStatus.SERVING, ServingStatus.NOT_SERVING, ServingStatus.SERVING);
```